provisioning_templates/provision/kickstart_default.erb

```diff
@@ -73,15 +73,40 @@ selinux --<%= host_param('selinux-mode') || host_param('selinux') || 'enforcing'
 keyboard <%= host_param('keyboard') || 'us' %>
 skipx
 
-<% subnet = @host.subnet -%>
-<% if subnet.respond_to?(:dhcp_boot_mode?) -%>
-<% dhcp = subnet.dhcp_boot_mode? && !@static -%>
+<%
+  # This template only supports either IPv4 or IPv6 but not both
+  if os_major >= 6
+    common_network_options = "--hostname #{@host} --device=#{@host.mac}"
+  else
+    common_network_options = "--hostname #{@host}"
+  end
+  if @host.subnet
+    subnet = @host.subnet
+    if @static
+      dhcp = false
+    elsif subnet.nil?
+      dhcp = true
+    else
+      dhcp = subnet.dhcp_boot_mode?
+    end
+-%>
+network --bootproto <%= dhcp ? 'dhcp' : "static --ip=#{@host.ip} --netmask=#{subnet.mask} --gateway=#{subnet.gateway} --nameserver=#{subnet.dns_servers.join(',')} --mtu=#{subnet.mtu.to_s}" %> <%= common_network_options -%>
+<%
+  elsif @host.subnet6
+    subnet = @host.subnet6
+    if @static
+      dhcp = false
+    elsif subnet.nil?
+      dhcp = true
+    else
+      dhcp = subnet.dhcp_boot_mode?
+    end
+-%>
+network --bootproto <%= dhcp ? 'dhcp' : "static --ipv6=#{@host.ip6}/#{subnet.cidr} --ipv6gateway=#{subnet.gateway} --nameserver=#{subnet.dns_servers.join(',')} --mtu=#{subnet.mtu.to_s}" %> <%= common_network_options -%>
```

Comment:
Not a blocker since previously IPv6 wasn't set up at all, but it would be nice if it could set up dual stack when both subnet6 and subnet are present.

Reply:
Only one interface can be marked as provisioning, there is no dualstack possible during provisioning and I don't want to ever hear about again. :-)

Reply:
Oh wait, one interface can be marked as provision however it can be assigned to both 4 and 6 subnets. Sure, that's possible in kickstart but the question is - which one to pick in PXE template (to actually boot from)? Any idea?

Reply:
My immediate thought is to use dual stack and let the OS figure it out. The documentation doesn't say a lot about repeating it, but I wonder if this works:
```
<% if subnet = @host.subnet -%>
network --bootproto <%= dhcp ? 'dhcp' : "static --ip=#{@host.ip} --netmask=#{subnet.mask} --gateway=#{subnet.gateway} --nameserver=#{subnet.dns_servers.join(',')} --mtu=#{subnet.mtu}" %> <%= network_options -%>
<% end -%>
<% if subnet6 = @host.subnet6 -%>
network --bootproto <%= dhcp ? 'dhcp' : "static --ipv6=#{@host.ip6}/#{subnet6.cidr} --ipv6gateway=#{subnet6.gateway} --nameserver=#{subnet6.dns_servers.join(',')} --mtu=#{subnet6.mtu}" %> <%= network_options -%>
<% end -%>
```

https://pykickstart.readthedocs.io/en/latest/kickstart-docs.html#id42 also mentions you may need to add `--noipv4` if there's no `subnet` but only `subnet6`. It may also be easier to construct it by building a large options array where you append options to.

Reply:
Yeah, this is definitely one concern. My bigger one is actually PXELinux - which stack to use to actually PXE boot from.

Reply:
For the record, ksvalidator disallows multiple `network` statements, therefore it must be only single line.